sponsorship_compassion/models/contracts_report.py

```diff
@@ -0,0 +1,226 @@
+    def _compute_boy(self):
+        for partner in self:
+            total = self.env['compassion.child'].search_count(
+                [('partner_id', '=', partner.id),
+                 ('gender', '=', 'M')])
+            if partner.is_church:
+                for member in partner.member_ids:
+                    total += self.env['compassion.child'].search_count(
+                        [('partner_id', '=', member.id),
+                         ('gender', '=', 'M')])
+            partner.sr_nb_boy = total
+
+    @api.multi
+    def _compute_girl(self):
+        for partner in self:
+            total = self.env['compassion.child'].search_count(
+                [('partner_id', '=', partner.id),
+                 ('gender', '=', 'F')])
+            if partner.is_church:
+                for member in partner.member_ids:
+                    total += self.env['compassion.child'].search_count(
+                        [('partner_id', '=', member.id),
+                         ('gender', '=', 'F')])
+            partner.sr_nb_girl = total
+
+    @api.multi
+    def _compute_time_scp(self):
+        def get_time_in_scp(_partner):
+            total = 0
+            for contract in _partner.sponsorship_ids.filtered(
+                    lambda s: s.is_active is True):
+                today = date.today()
+                contract_start_date = fields.Date.from_string(
+                    contract.start_date)
+                delta = today - contract_start_date
+
+                nb_weeks = delta.days / 7
+                country = contract.child_id.field_office_id
+                total += nb_weeks * country.icp_hours_week
+            return total
+
+        for partner in self:
+            time_to_scp = get_time_in_scp(partner)
+
+            if partner.is_church:
+                for member in partner.member_ids:
+                    time_to_scp += get_time_in_scp(member)
+            partner.sr_time_scp = time_to_scp
+
+    @api.multi
+    def _compute_meal(self):
+        def get_nb_meal(_contract):
+            today = date.today()
+            contract_start_date = fields.Date.from_string(
+                _contract.start_date)
+            delta = today - contract_start_date
+
+            nb_weeks = delta.days / 7
+            country = contract.child_id.field_office_id
+            return nb_weeks * country.icp_meal_week
+
+        for partner in self:
+            sr_meal = 0
+            for contract in partner.sponsorship_ids:
+                sr_meal += get_nb_meal(contract)
+
+            if partner.is_church:
+                for member in partner.member_ids:
+                    for contract in member.sponsorship_ids:
+                        sr_meal += get_nb_meal(contract)
+            partner.sr_meal = sr_meal
+
+    @api.multi
+    def _compute_sr_payment(self):
+        def get_sum_invoice(_partner):
+            invoices = self.env['account.invoice.line'].search([
+                ('partner_id', '=', _partner.id),
```

Comment:
Add filter `('type', '=', 'out_invoice')` to avoid taking supplier invoices.
Add filter `('state', '=', 'paid')` otherwise you will include open invoices.
Add filter `('invoice_type', 'in', ['gift', 'sponsorship', 'fund'])` to exclude all invoices that would not be a donation.